labs/LanguageBasics/LanguageBasics.md

```diff
@@ -2,39 +2,281 @@
+elements of three bits each). So `[~1, 1] == [6, 1] == [6, 4 + 5]`.
+
+The `0b0100` in the second example needs four bits, so both sides have
+type `[2][4]`. In this case `[~1, 1] == [14, 1]` while `[6, 4 + 5] ==
+[6, 9]` so equality fails.
+
+_**It is important to be precise about the widths of things!**_
+
+Comparisons are lexicographic on sequences of numbers.
+
+```sh
+labs::LanguageBasics::LanguageBasics> [1, 2] < [1, 3]
+True
+labs::LanguageBasics::LanguageBasics> [1, 2] < [1, 2]
+False
+```
+
+### Shifts: `<<`, `>>`, `<<<` and `>>>`
+#### Signed version: `>>$`
+
+```sh
+labs::LanguageBasics::LanguageBasics> 0xa5a << 4
+0x5a0
+labs::LanguageBasics::LanguageBasics> 0xa5a << 12
+0x000
+labs::LanguageBasics::LanguageBasics> 0xa5a <<< 16
+0x5aa
+```
+
+### Indexing and slicing: `@`, `!`, `@@` and `!!`
+
+```sh
+labs::LanguageBasics::LanguageBasics> "cat" @ 0
+'c'
+labs::LanguageBasics::LanguageBasics> "dog" @@ [2, 1, 1, 0, 0, 1, 2]
+"gooddog"
+labs::LanguageBasics::LanguageBasics> "cow" ! 0
+'w'
+```
+
+### Concatenation: `#`
+
 ```sh
-labs::LanguageBasics::LanguageBasics> :t abs
+labs::LanguageBasics::LanguageBasics> "dog" # "cow" // Moof!
+"dogcow"
+```
+
+### Shortcutting logical:: `/\`, `\/` and `==>`
+
+These are most often used in property statements. `/\` is "and", `\/` is
+"or" and `==>` is "implies". They have very low precedence.
+
+```sh
+labs::LanguageBasics::LanguageBasics> 1 == 5 \/ 5 == 5
+True
+labs::LanguageBasics::LanguageBasics> False ==> 1 == 5 /\ 1 != 5
+True
+```
+
+
+Common Primitives
+-----------------
+
+Cryptol's `:help` command will provide a brief description of the
+primitives in the section by issuing `:help ` followed
+by the name of the primitive.
+
+### Collections of all `False` or all `True` bits
+
+* `0` is a sequence of `False` bits whose type is determinted by the
+context.
+  ```sh
+  labs::LanguageBasics::LanguageBasics> 0 : [12]
+  0x000
+  ```
+* `zero` is an arbitrary collection of `False` bits whose type
+is determinted by the context.
+  ```sh
+  labs::LanguageBasics::LanguageBasics> zero: ([8], [4])
+  (0x00, 0x0)
+  ```
+  Here we produce an order pair of a 0 octext and a 0 nibble.
+* `~0` and `~zero` produce all `True` bits correspondingly.
+
+
+### List manipulation: `take`, `drop`, `tail`, `last` and `reverse`
+```
+labs::LanguageBasics::LanguageBasics> take "dogcow" : [3][8]
+"dog"
+labs::LanguageBasics::LanguageBasics> drop`{2} [2, 3, 5, 7, 11]
+[5, 7, 11]
+labs::LanguageBasics::LanguageBasics> tail [0, 1, 1]
+[1, 1]
+labs::LanguageBasics::LanguageBasics> last [2, 3, 5, 7, 11]
+11
+labs::LanguageBasics::LanguageBasics> reverse [0, 0, 1]
+[1, 0, 0]
+```
+
+Of course the sizes of lists have to be big enough. Also, notice that
+last (which is equivalent to `!0`) returns an element while the others
+return lists.
+
+Often in a Cryptol program, the context will determine the shapes of
+sequences, so that the type annonations (`: [3][8]` and ``{2}` above)
+will be unnecesarry.
```

Comment:
The double back-tick messes with the formatting...  Perhaps just don't delinate the `{2} reference.  Also, "annonations" --> "annotations"

Reply:
I think you can use three backticks, like so: ``` `{2} ```

Reply:
I think `` `{2} `` does the right think. Will not resolve this until I see it is so.

Reply:
fixed spelling